Shell/main.cpp

```diff
@@ -147,6 +147,43 @@ static int sh_unset(int argc, char** argv)
     return 0;
 }
 
+static String expanded_tilde_expression_from(char* arg)
+{
+    int len = strlen(arg);
+    ASSERT(len);
+
+    StringBuilder login_name;
+    int first_slash_index = len;
+    for (int i = 1; i < len; ++i) {
+        if (arg[i] == '/') {
+            first_slash_index = i;
+            break;
+        }
+        login_name.append(arg[i]);
+    }
+
+    StringBuilder path;
+    for (int i = first_slash_index; i < len; ++i)
+        path.append(arg[i]);
+
+    if (login_name.length() == 0) {
+        char* home = getenv("HOME");
+        if (!home) {
+            auto passwd = getpwuid(getpid());
+            ASSERT(passwd && passwd->pw_dir);
+            return String::format("%s/%s", passwd->pw_dir, path.to_string().characters());
+        }
+        return String::format("%s/%s", home, path.to_string().characters());
+    }
+
+    auto passwd = getpwnam(login_name.to_string().characters());
+    if (!passwd)
+        return String(arg);
```

Comment:
Why are we returning arg?

Reply:
Because it's the original path that we failed to tilda-expand, so we return it unmodified. Of course the name `arg` is not ideal..?

Reply:
I will change to a better name